pkg/start/start.go

```diff
@@ -195,96 +192,103 @@ func (o *Options) run(ctx context.Context, controllerCtx *Context, lock *resourc
 	controllerCtx.OpenshiftConfigManagedInformerFactory.Start(informersDone)
 	controllerCtx.InformerFactory.Start(informersDone)
 
-	resultChannelCount++
-	go func() {
-		defer utilruntime.HandleCrash()
-		leaderelection.RunOrDie(postMainContext, leaderelection.LeaderElectionConfig{
-			Lock:            lock,
-			ReleaseOnCancel: true,
-			LeaseDuration:   o.leaderElection.LeaseDuration.Duration,
-			RenewDeadline:   o.leaderElection.RenewDeadline.Duration,
-			RetryPeriod:     o.leaderElection.RetryPeriod.Duration,
-			Callbacks: leaderelection.LeaderCallbacks{
-				OnStartedLeading: func(_ context.Context) { // no need for this passed-through postMainContext, because goroutines we launch inside will use runContext
-					launchedMain = true
-					if o.ListenAddr != "" {
+	if !cache.WaitForCacheSync(informersDone, controllerCtx.CVO.GetCacheSynced()...) {
+		err := postMainContext.Err()
+		klog.V(2).Infof("Caches never synchronized: %v", err)
+	} else if err := controllerCtx.CVO.InitializeFromPayload(ctx, restConfig, burstRestConfig); err != nil {
+		klog.V(2).Infof("Initialize from payload: err=%v", err)
+	} else {
```

Comment:
nit: instead of nesting all of this, can we have the earlier blocks of this `if`/`else` return their errors instead of logging them and falling through?

Reply:
used Fatalf